### ISSUE TITLE
DRBD device not deleted in case of problem during drbd setup

### ISSUE THREAD

**Issue by Ganeti-Issues-Migrator** [ganeti/ganeti#130]
Originally reported of Google Code with ID 74.

```
Testing the last drbd8 module version (in debian sid), I tried to create a
vm between node1 and node2 :
- node2 has a buggy drbd8 module
- node1 has a working one (8.3.2-3)

When I tried to create a vm , with :

gnt-instance add -t drbd -s 5g -B memory=128 -n node2.foo.tld:node1.foo.tld
-o debootstrap --net 0:mac=52:54:43:c1:9e:7f gnt1.tld

I got a kernel oops on node2, and the drbd device can't be created. 
After a long time (+10min), the ganeti job still was "running", a
"/etc/init.d/ganeti restart" on the cluster master stopped it.

Anyway, I still have a /dev/drbd11 device on node1, created before the
crash, which is not known by ganeti, so that if I want to create another
vm, I get errors like :

Can't create block device <DRBD8(hosts=z2-6.host.gnt/1-z2-5.host.gnt/11,
port=11191, configured as 10.10.0.5:11191 10.10.0.6:11191,
backend=<LogicalVolume(/dev/all/bc669074-d0db-4358-87b1-187797e451a2.disk0_data,
not visible,            
size=10240m)>,
metadev=<LogicalVolume(/dev/all/bc669074-d0db-4358-87b1-187797e451a2.disk0_meta,
not visible, size=128m)>, visible as /dev/disk/0, size=10240m)> on node
z2-5.host.gnt for instance wetball.mekensleep.vm.gnt: Can't create   
block device: drbd11: minor is already in use at Create() time  

It seems that ganeti doesn't handle rollover in case of failure during 
gnt-instance add.

Maxence
```
Originally added on 2009-10-20 10:35:43 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
The "/etc/init.d/ganeti restart" kills all current jobs without possibility of
cleanup. Ganeti cannot in this scenario recover all resources, as we don't keep a
journal of operations. Hmm, a journal sounds interesting.

But in any case, after a restart, if you do a "gnt-cluster verify", it will tell you
that unused minor 11 is in use, and you should deconfigure it manually (for now at
least).
```
Originally added on 2009-10-26 03:23:16 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
The 2.3 release will hopefully be able to automatically cleanup resources that are invalid but in use.
```
Originally added on 2010-07-28 23:24:22 +0000 UTC.

Added Labels: Type-Enhancement Priority-Medium
Changed State: Accepted
Added to Milestone: Release2.3

**Comment by Ganeti-Issues-Migrator** [comment]
```
De-targetting these issues, were not implemented in 2.3.
```
Originally added on 2011-02-18 13:43:21 +0000 UTC.

Added to Milestone: Unplanned
Removed from Milestone: Release2.3

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2011-08-05 14:45:42 +0000 UTC.

Removed from Milestone: Unplanned

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2012-10-22 08:57:07 +0000 UTC.

Changed State: Released

**Comment by Ganeti-Issues-Migrator** [comment]
```
Undo wrong status update to 'released'.
```
Originally added on 2012-10-22 09:03:08 +0000 UTC.

Changed State: New

**Comment by Ganeti-Issues-Migrator** [comment]
```
In 2.7 we made as much effort as possible to cleanup resources that were created by mistake. Of course not everything can be fixed, but I think this bug should be considered addressed there.
```
Originally added on 2013-05-08 08:06:26 +0000 UTC.

Changed State: Fixed
Added to Milestone: Release2.7

**Comment by Ganeti-Issues-Migrator** [comment]
```
Marking 2.7 bugs fixed, as 2.7.0~rc1 was released, and should be ready to be tested in production environments.
```
Originally added on 2013-05-09 05:42:48 +0000 UTC.

Changed State: Released
